src/createWidgetBase.ts

```diff
@@ -156,7 +147,7 @@ function generateProperties(instance: Widget<WidgetState, WidgetProperties>, pre
 	};
 
 	changedPropertyKeys.forEach((key) => {
-		changedProperties.currentProperties[key] = instance.properties[key];
+		changedProperties.currentProperties[key] = (<any> instance.properties)[key];
```

Comment:
This is a hammer at the moment as I had to remove the index type as they do not work with Partials well.